test/destChain_test/Prover_Axelar_test.ts

```diff
@@ -0,0 +1,365 @@
+      it("Should successfully handle complete flow from deal client to receiver", async function () {
+        // Setup chain configuration
+        const chainId = 1;
+        const chainName = "filecoin-2";
+        await dealClient.setDestinationChains(
+          [chainId],
+          [chainName],
+          [await axelarBridge.getAddress()]
+        );
+
+        // Setup bridge configuration
+        await axelarBridge.setSenderReceiver(
+          await dealClient.getAddress(),
+          await debugReceiver.getAddress()
+        );
+
+        // Prepare test data
+        const commP = ethers.hexlify(ethers.randomBytes(32));
+        const providerAddr = ethers.hexlify(ethers.randomBytes(32));
+        const gasFunds = ethers.parseEther("0.1");
+
+        // Add gas funds
+        await dealClient.addGasFunds(providerAddr, { value: gasFunds });
+
+        // Trigger debug call
+        await expect(
+          dealClient.debug_call(commP, providerAddr, gasFunds, chainId)
+        )
+          .to.emit(mockGasService, "GasPaid")
+          .and.to.emit(mockGateway, "ContractCall");
+
+        // Verify gas funds were deducted
+        expect(await dealClient.providerGasFunds(providerAddr)).to.equal(0);
+      });
+
+      it("Should handle cross-chain attestation flow", async function () {
+        const chainId = 2; // Different chain than the current one
+        const chainName = "filecoin-2";
+        await dealClient.setDestinationChains(
+          [chainId],
+          [chainName],
+          [await axelarBridge.getAddress()]
+        );
+
+        const commP = ethers.hexlify(ethers.randomBytes(32));
+        const providerAddr = ethers.hexlify(ethers.randomBytes(32));
+        const gasFunds = ethers.parseEther("0.2");
+
+        await dealClient.addGasFunds(providerAddr, { value: gasFunds });
+
+        // Create and encode a deal proposal
+        const pieceCID = {
+          "/": Buffer.from(commP.slice(2), "hex").toString("base64"),
+        };
+
+        const dealProposal = {
+          PieceCID: pieceCID,
+          PieceSize: 1n,
+          VerifiedDeal: false,
+          Client: Buffer.from(owner.address.slice(2), "hex"),
+          Provider: Buffer.from(providerAddr.slice(2), "hex"),
+          Label: Buffer.from(
+            ethers.AbiCoder.defaultAbiCoder().encode(["uint256"], [chainId])
+          ),
+          StartEpoch: 1n,
+          EndEpoch: 2n,
+          StoragePricePerEpoch: 1n,
+          ProviderCollateral: 1n,
+          ClientCollateral: 1n,
+        };
+
+        const marketDealNotifyParams = {
+          dealProposal: Buffer.from(cbor.encode(dealProposal)),
+          dealId: 1n,
+        };
+
+        const params = Buffer.from(cbor.encode(marketDealNotifyParams));
+
+        // Impersonate market actor and notify deal
+        const impersonatedSigner = await ethers.getImpersonatedSigner(
+          MARKET_ACTOR_ADDRESS
+        );
+        await owner.sendTransaction({
```

Comment:
What does this function do? I do not see how is it calling dealNotify()

Reply:
The impersonatedSigner part is used to simulate a call from the Market Actor (0xff00000000000000000000000000000000000005) since dealNotify() can only be called via handle_filecoin_method by the Market Actor. The test tries to create a deal proposal and call handle_filecoin_method with MARKET_NOTIFY_DEAL_METHOD_NUM (4186741094), which internally calls dealNotify().

However, I notice two issues in my test:
1. The crucial part that actually calls dealNotify() is commented out (the handle_filecoin_method call)
2. The test is still failing due to CBOR encoding issues that I haven't resolved yet